# Collecting Article Data from New York Time's Archive

Over 150 years of NYT articles can be collected from the Archive API. This notebook extracts some useful information for some analysis for a period of time.

#### Importing necessary libraries

In [1]:
import pandas as pd
import json
import configparser
import requests
import time

#### Reading the API key from Configparser

In [2]:
configs = configparser.ConfigParser()
configs.read('News_API.ini')
API_KEY = configs['news_API']['nytimes_archive']

#### Simple Data Extractor

The NYT allows to extract one month of archive data in one call and the data is in JSON file format. A simple function is created to extract one JSON file contains one month of articles.

In [3]:
def send_request(yr, month, api_key):
  '''Sends a request to the NYT Archive API for a given year and month, and receive the response'''
    
  base_url = 'https://api.nytimes.com/svc/archive/v1'
  full_url = base_url + '/' + yr + '/' + month + '.json?api-key=' + api_key
    
  try:
    response = requests.get(full_url).json()
  except Exception:
    return None
  time.sleep(6)
  return response

Sample output of `send_request()` function.

In [4]:
nyt_articles = send_request(str(2021), str(6), API_KEY)
len(nyt_articles['response']['docs'])

4260

Information about 4260 articles comprised in the June 2021 archive. First dictionary item is printed to see what data is available for further extraction and analysis.

In [5]:
nyt_articles['response']['docs'][0]

{'abstract': 'In Israel’s newspapers — as fractured as its electorate — the opinions on the possible fall of Prime Minister Benjamin Netanyahu were as emphatic as they were varied.',
 'web_url': 'https://www.nytimes.com/2021/05/31/world/middleeast/israeli-media-netanyahu-bennett.html',
 'snippet': 'In Israel’s newspapers — as fractured as its electorate — the opinions on the possible fall of Prime Minister Benjamin Netanyahu were as emphatic as they were varied.',
 'lead_paragraph': 'After more than 12 consecutive years at the helm of Israel’s government, Prime Minister Benjamin Netanyahu’s hold on power may be coming to a close.',
 'source': 'The New York Times',
 'multimedia': [{'rank': 0,
   'subtype': 'xlarge',
   'caption': None,
   'credit': None,
   'type': 'image',
   'url': 'images/2021/05/31/world/31israel-reaction/merlin_188519604_9d0ecaa9-cca2-4523-ae7c-63df787502c5-articleLarge.jpg',
   'height': 400,
   'width': 600,
   'subType': 'xlarge',
   'crop_name': 'articleLarge',